node/src/types/block/block_payload.rs

```diff
@@ -56,6 +56,14 @@ impl BlockPayload {
         self.transfer.iter()
     }
 
+    /// Returns the hashes and approvals of the non-transfer transactions within the block.
+    pub fn non_transfer(&self) -> impl Iterator<Item = &TransactionHashWithApprovals> {
```

Comment:
As per the comment on block validation, this method should probably be replaced with three others, one per category.

Reply:
Yes, it is updated in 9b80304356f3bebccfb095308af1fde508cb8324